src/main/kotlin/org/jitsi/nlj/transform/node/ToggleablePcapWriter.kt

```diff
@@ -25,38 +25,31 @@ import org.jitsi.nlj.PacketInfo
 import org.jitsi.utils.logging2.Logger
 import java.lang.IllegalStateException
 import java.util.Date
+import java.util.concurrent.atomic.AtomicReference
 
 class ToggleablePcapWriter(
-    val parentLogger: Logger,
-    val prefix: String
+    private val parentLogger: Logger,
+    private val prefix: String
 ) {
-    private var pcapWriter: PcapWriter? = null
-
-    private val pcapLock = Any()
+    private val pcapWriter = AtomicReference<PcapWriter?>(null)
 
     fun enable() {
         if (!enabled) {
             throw IllegalStateException("PCAP capture is disabled in configuration")
         }
-        synchronized(pcapLock) {
-            if (pcapWriter == null) {
-                pcapWriter = PcapWriter(parentLogger, "/tmp/$prefix-${Date().toInstant()}.pcap")
-            }
-        }
+
+        pcapWriter.compareAndSet(null, PcapWriter(parentLogger, "/tmp/$prefix-${Date().toInstant()}.pcap"))
```

Comment:
Is it okay that this will create a new instance every time, even when it doesn't update?

Reply:
This will only happen if the current value is `null`.

Reply:
Yes, but it's still not great. It depends on the vals in `PcapWriter` being lazy.

Reply:
Ahh, I see what you mean.  Yes that's a good point, actually.  I don't like that either.  I'll think about it.

Reply:
It would be nice if you could do `compareAndSet(null) { PcapWriter(...) }`.

Reply:
Ok, I reverted it back to use the lock.

Reply:
> It would be nice if you could do `compareAndSet(null) { PcapWriter(...) }`.

Yeah I was hoping/looking for exactly that, but `AtomicReference` doesn't support it and my quick search didn't turn up any alternative.